src/commands/flinkArtifacts.test.ts

```diff
@@ -40,149 +44,189 @@ describe("flinkArtifacts", () => {
-
-    sandbox.stub(artifactUploadForm, "artifactUploadQuickPickForm").resolves(params);
-    sandbox.stub(uploadArtifact, "getPresignedUploadUrl").resolves(uploadUrl);
-    sandbox.stub(uploadArtifact, "handleUploadToCloudProvider").resolves();
-
-    sandbox
-      .stub(uploadArtifact, "uploadArtifactToCCloud")
-      .rejects(createResponseError(500, "Oops, something went wrong", ""));
-
-    const showErrorStub = getShowErrorNotificationWithButtonsStub(sandbox);
-
-    await uploadArtifactCommand();
-
-    sinon.assert.calledOnce(showErrorStub);
-    sinon.assert.calledWithMatch(
-      showErrorStub,
-      "Please make sure that you provided a valid JAR file",
-    );
-  });
-
-  it("should error for other status codes", async () => {
-    const params = { ...mockParams };
-    const uploadUrl = { ...mockPresignedUrlResponse };
-
-    sandbox.stub(artifactUploadForm, "artifactUploadQuickPickForm").resolves(params);
-    sandbox.stub(uploadArtifact, "getPresignedUploadUrl").resolves(uploadUrl);
-    sandbox.stub(uploadArtifact, "handleUploadToCloudProvider").resolves();
-
-    sandbox.stub(uploadArtifact, "uploadArtifactToCCloud").rejects(
-      createResponseError(
-        400,
-        "Custom Bad Request",
-        JSON.stringify({
-          errors: [
-            {
-              detail: "Custom Bad Request Body",
-            },
-          ],
-        }),
-      ),
-    );
-
-    const showErrorStub = getShowErrorNotificationWithButtonsStub(sandbox);
-
-    await uploadArtifactCommand();
-
-    sinon.assert.calledOnce(showErrorStub);
-    sinon.assert.calledWithMatch(
-      showErrorStub,
-      "Failed to upload artifact: Custom Bad Request Body",
-    );
-  });
-
-  it("should send the create artifact request to Confluent Cloud", async () => {
-    const mockUploadId = "12345";
-    const mockCreateResponse = {
-      display_name: "test-artifact",
-      id: "artifact-123",
-      environment: "env-123456",
-      region: "australiaeast",
-      cloud: "Azure",
-    };
-
-    sandbox.stub(artifactUploadForm, "artifactUploadQuickPickForm").resolves(mockParams);
-    sandbox.stub(uploadArtifact, "getPresignedUploadUrl").resolves(mockPresignedUrlResponse);
-    const handleUploadStub = sandbox.stub(uploadArtifact, "handleUploadToCloudProvider").resolves();
-    const createArtifactStub = sandbox
-      .stub(uploadArtifact, "uploadArtifactToCCloud")
-      .resolves(mockCreateResponse);
-    sandbox.stub(vscode.window, "showInformationMessage");
-
-    await uploadArtifactCommand();
-
-    sinon.assert.calledOnce(handleUploadStub);
-    sinon.assert.calledWithExactly(handleUploadStub, mockParams, mockPresignedUrlResponse);
-
-    sinon.assert.calledOnce(createArtifactStub);
-    sinon.assert.calledWithExactly(createArtifactStub, mockParams, mockUploadId);
+  describe("handleWithProgressForUploadArtifact", () => {
+    it("should report progress correctly through all steps", async () => {
+      const mockCreateResponse = {
```

Comment:
I almost want to put this and `mockProgress` in the top-level describe by default now... but it's the first instance